modules/util/util.js

```diff
@@ -542,3 +542,65 @@ export function utilUnicodeCharsCount(str) {
 export function utilUnicodeCharsTruncated(str, limit) {
     return Array.from(str).slice(0, limit).join('');
 }
+
+// Simple polyfill for String.prototype.padStart().
+function padStart(str, width, pad) {
+    var padded = Array(width);
+    padded.fill(pad);
+    padded = padded.join('');
+    return padded.substr(0, width - str.length) + str;
+}
+
+// Returns an object containing the given date string normalized as an ISO 8601 date string and parsed as a Date object.
+// Date components are padded for compatibility with tagging conventions.
+// Dates such as January 0, February 31, and Duodecember 1 are wrapped to make more sense.
+export function utilNormalizeDateString(raw) {
+    var date;
+
+    // Enforce the same date formats supported by DateFunctions-plpgsql and decimaldate-python.
+    var dateRegex = /^(-)?(\d+)(?:-(\d\d?)(?:-(\d\d?))?)?$/;
+    var match = raw.match(dateRegex);
+    if (match !== null) {
+        // Manually construct a Date.
+        // Passing the string directly into the Date constructor would throw an error on negative years.
+        date = new Date(0);
+        date.setUTCFullYear(parseInt((match[1] || '') + match[2], 10));
+        if (match[3]) date.setUTCMonth(parseInt(match[3], 10) - 1); // 0-based
+        if (match[4]) date.setUTCDate(parseInt(match[4], 10));
+    } else {
+        // Fall back on whatever the browser can parse into a date.
+        date = new Date(raw);
```

Comment:
This fallback is what allows for some natural language syntax to be parsed successfully. The specific supported natural language formats depends on the browser, language settings, etc., so it’s only a best-effort fallback. Anything produced by this code path still results in an error presented to the user, but there’s a suggestion to reformat the tag value to what the `Date` constructor interprets the value as.